Lib/test/datetimetester.py

```diff
@@ -1106,6 +1106,65 @@ def test_delta_non_days_ignored(self):
         dt2 = dt - delta
         self.assertEqual(dt2, dt - days)
 
+    def test_strptime(self):
+        string = '2004-12-01'
+        format = '%Y-%m-%d'
+        expected = _strptime._strptime_datetime_date(date, string, format)
+        got = date.strptime(string, format)
+        self.assertEqual(expected, got)
+        self.assertIs(type(expected), date)
+        self.assertIs(type(got), date)
+
+        # bpo-34482: Check that surrogates are handled properly.
+        inputs = [
+            ('2004-12\ud80001', '%Y-%m\ud800%d'),
+            ('2004\ud80012-01', '%Y\ud800%m-%d'),
+        ]
+        for string, format in inputs:
+            with self.subTest(string=string, format=format):
+                expected = _strptime._strptime_datetime_date(date, string,
+                                                             format)
+                got = date.strptime(string, format)
+                self.assertEqual(expected, got)
+
```

Comment:
We should probably test that we fail when passed time components:

```suggestion

def test_strptime_date_with_time_error(self):
    formats = [
        ('12', '%H'),
        ('03', '%M'),
        ('12', '%S'),
        ('12:30:14.531', '%H:M:%S.%f'),
        ('2004-02-03 04:11:23.145', '%Y-%m-%d %H:%M:%S.%f'), 
        ('2004-02 12', '%Y-%m %H'),
    ]

    for dtstr, format in formats:
        with self.subTest(format=format):
            with self.assertRaises(ValueError):
                self.theclass.strptime(dtstr, format)
```